internal/provider/resources/work_pool_access.go

```diff
@@ -0,0 +1,343 @@
+	"github.com/prefecthq/terraform-provider-prefect/internal/provider/customtypes"
+	"github.com/prefecthq/terraform-provider-prefect/internal/provider/helpers"
+)
+
+var _ = resource.ResourceWithConfigure(&WorkPoolAccessResource{})
+
+type WorkPoolAccessResource struct {
+	client api.PrefectClient
+}
+
+type WorkPoolAccessResourceModel struct {
+	AccountID   customtypes.UUIDValue `tfsdk:"account_id"`
+	WorkspaceID customtypes.UUIDValue `tfsdk:"workspace_id"`
+
+	WorkPoolName types.String `tfsdk:"work_pool_name"`
+
+	ManageActorIDs types.List `tfsdk:"manage_actor_ids"`
+	RunActorIDs    types.List `tfsdk:"run_actor_ids"`
+	ViewActorIDs   types.List `tfsdk:"view_actor_ids"`
+	ManageTeamIDs  types.List `tfsdk:"manage_team_ids"`
+	RunTeamIDs     types.List `tfsdk:"run_team_ids"`
+	ViewTeamIDs    types.List `tfsdk:"view_team_ids"`
+}
+
+// NewWorkPoolAccessResource returns a new WorkPoolAccessResource.
+//
+//nolint:ireturn // required by Terraform API
+func NewWorkPoolAccessResource() resource.Resource {
+	return &WorkPoolAccessResource{}
+}
+
+// Metadata returns the resource type name.
+func (r *WorkPoolAccessResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_work_pool_access"
+}
+
+// Configure initializes runtime state for the resource.
+func (r *WorkPoolAccessResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(api.PrefectClient)
+	if !ok {
+		resp.Diagnostics.Append(helpers.ConfigureTypeErrorDiagnostic("resource", req.ProviderData))
+
+		return
+	}
+
+	r.client = client
+}
+
+// Schema returns the resource schema.
+func (r *WorkPoolAccessResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	defaultEmptyList, _ := basetypes.NewListValue(types.StringType, []attr.Value{})
+
+	resp.Schema = schema.Schema{
+		Description: "The resource `work_pool_access` represents a connection between an accessor " +
+			"(User, Service Account or Team) with a Work Pool. This resource specifies an actor's access level " +
+			"to a specific Work Pool in the Account. " +
+			"For more information, see [object access control lists](https://docs.prefect.io/v3/manage/cloud/manage-users/object-access-control-lists).",
+		Version: 0,
+		Attributes: map[string]schema.Attribute{
+			"account_id": schema.StringAttribute{
+				Optional:    true,
+				Description: "Account ID (UUID)",
+				CustomType:  customtypes.UUIDType{},
+			},
+			"workspace_id": schema.StringAttribute{
+				Optional:    true,
+				Description: "Workspace ID (UUID)",
+				CustomType:  customtypes.UUIDType{},
+			},
+			"work_pool_name": schema.StringAttribute{
+				Required:    true,
+				Description: "The name of the Work Pool",
+			},
+			"manage_actor_ids": schema.ListAttribute{
```

Comment:
This is all repeated from `deployment_access` and `workspace_access`. We can de-duplicate a lot of that code. I'll make a followup issue to do that.

Reply:
https://linear.app/prefect/issue/PLA-1180/de-duplicate-code-for-access-resources